include/boost/unordered/detail/foa/node_handle.hpp

```diff
@@ -153,7 +182,17 @@ struct node_handle_base
       }
     }
 
-    allocator_type get_allocator()const noexcept{return al();}
+    allocator_type get_allocator()const
+    {
+#if defined(BOOST_GCC)
+      /* GCC lifetime analysis incorrectly warns about uninitialized
+       * allocator object under some circumstances.
+       */
+      if(empty())__builtin_unreachable();
```

Comment:
This seems fishy to me. Is calling `get_allocator` undefined behavior when `empty()`? Is the removal of the `noexcept` on purpose?

Reply:
* `get_allocator` on an empty node handle is UB, see [here](https://eel.is/c++draft/container.node#observers-11).
* I removed `noexcept` on purpose cause the standard does not have it (presumably because of the Lakos rule on narrow-contract functions). This removal is not materially connected to the `__builtin_unreachable` addition, really.
* The `__builtin_unreachable` addition is due to GCC incorrectly, and not systematically, warning about potentially uninitialized node handles [here](https://github.com/boostorg/unordered/blob/bd5eae3c1a97d5557e8a0b4bfc513d1eed3c14bd/test/unordered/node_handle_allocator_tests.cpp#L130) and [here](https://github.com/boostorg/unordered/blob/bd5eae3c1a97d5557e8a0b4bfc513d1eed3c14bd/test/unordered/node_handle_allocator_tests.cpp#L138): adding  `__builtin_unreachable` helped _but not always_, so I was forced to also add [this](https://github.com/boostorg/unordered/blob/bd5eae3c1a97d5557e8a0b4bfc513d1eed3c14bd/test/unordered/node_handle_allocator_tests.cpp#L7-L13). I don't see any problem with foolproofing `node_handle_base::get_allocator` as I've done, but if it looks smelly/ugly I can just remove it, of course --the GCC pragma in `node_handle_allocator_tests.cpp` has us covered.